buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTestingPlugin.kt

```diff
@@ -67,22 +69,24 @@ class DistributionTestingPlugin : Plugin<Project> {
         }
     }
 
-    private fun Project.setDedicatedTestOutputDirectoryPerTask(distributionTest: DistributionTest) {
+    private
+    fun Project.setDedicatedTestOutputDirectoryPerTask(distributionTest: DistributionTest) {
         distributionTest.reports.junitXml.destination = File(java.testResultsDir, distributionTest.name)
         // TODO Confirm that this is not needed
         afterEvaluate {
             distributionTest.reports.html.destination = file("${the<ReportingExtension>().baseDir}/$name")
         }
     }
 
-    private fun Project.configureGradleTestEnvironment(distributionTest: DistributionTest): Unit = distributionTest.run {
+    private
+    fun Project.configureGradleTestEnvironment(distributionTest: DistributionTest): Unit = distributionTest.run {
         gradleInstallationForTest.run {
             val intTestImage: Sync by tasks
-            val toolingApiShadedJar: Zip by rootProject.project(":toolingApi").tasks
+            val toolingApiShadedJar: ShadedJar by rootProject.project(":toolingApi").tasks
             gradleHomeDir.set(dir { intTestImage.destinationDir })
             gradleUserHomeDir.set(rootProject.layout.projectDirectory.dir("intTestHomeDir"))
             daemonRegistry.set(rootProject.layout.buildDirectory.dir("daemon"))
-            toolingApiShadedJarDir.set(dir { toolingApiShadedJar.destinationDir })
+            toolingApiShadedJarDir.set(dir { toolingApiShadedJar.jarFile.get().asFile })
```

Comment:
This doesn't make sense. a) we are connecting two providers, but we are doing it via a closure, b) we are setting a directory property to a directory that points to a file?!

Reply:
How is b) not result in an error somewhere?

Reply:
I actually want to set this to the `parentFile` of the `jarFile`. I don't know how the tests passed with this code...

Reply:
Ahh, I set it in the wrong place - `DistributionTest`...

Reply:
But you are right, validation should catch this!

Reply:
Fixed.